framework/serialization/serialize.cpp

```diff
@@ -0,0 +1,385 @@
+#define PUGIXML_NO_XPATH
+#define PUGIXML_HEADER_ONLY
+#include "dependencies/pugixml/src/pugixml.hpp"
```

Comment:
As we're including the pugixml.hpp header lots of times with PUGIXML_HEADER_ONLY it probably makes sense to disable that and add pugixml.cpp to the build instead.